tiledb/sm/rest/curl.cc

```diff
@@ -679,6 +684,8 @@ Status Curl::check_curl_errors(
     }
 
     return Status_RestError(msg.str());
+  } else if (curl_code != CURLE_OK) {
```

Comment:
This block is unreachable because this condition matches the first part of the `||` clause in the `if` block.